src/systems.rs

```diff
@@ -337,18 +335,41 @@ pub(crate) fn particle_transform(
                 time.raw_delta_seconds()
             };
 
-            // Apply acceleration
-            velocity.0 += particle.acceleration * delta_time;
+            // inititalize precalculated values
+            let mut ppv = PrecalculatedParticleVariables::new();
 
-            // Apply drag
-            let current_drag = particle.drag.at_lifetime_pct(lifetime_pct);
-            if current_drag > 0.0 {
-                let drag_force = velocity.0.length_squared() * current_drag * delta_time;
-                let drag_force = -velocity.0.normalize() * drag_force;
-                velocity.0 += drag_force;
-            }
+            // Apply velocity modifiers to velocity
+            for modifier in &particle.velocity_modifiers {
+                use VelocityModifier::{Constant, Drag, Noise, ScalarAcceleration};
+                match modifier {
+                    Constant(v) => {
+                        velocity.0 += *v * delta_time;
+                    }
+
+                    ScalarAcceleration(v) => {
+                        let direction = ppv.get_particle_direction(&velocity.0);
+                        velocity.0 += v.at_lifetime_pct(lifetime_pct) * direction * delta_time;
+                    }
+
+                    Drag(v) => {
+                        let current_drag = v.at_lifetime_pct(lifetime_pct);
+                        if current_drag > 0.0 {
+                            let drag_force =
+                                ppv.get_particle_sqr_speed(&velocity.0) * current_drag * delta_time;
+                            let direction = ppv.get_particle_direction(&velocity.0);
+                            velocity.0 -= direction * drag_force;
+                        }
+                    }
 
-            // Apply velocity to translation
+                    Noise(n) => {
+                        let offset = n.sample(
+                            Vec2::new(transform.translation.x, transform.translation.y),
+                            time.elapsed_seconds_wrapped(),
```

Comment:
I think we should make the same distinction here as we do with delta_time above, using `elapsed_seconds_wrapped()` if we're respecting time scaling and `raw_elapsed_seconds_wrapped()` if we're not, just for consistencies sake.